### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SCREENREADER]: Screen reader users must be informed that the representative documentation section is incomplete

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va-forms-system-core#276]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to know if I must complete an additional section on the review page if I change a selection.

## Details

If a screen reader user decides to switch their choice to "Yes, I want to upload my document now," they are not provided any indication that it must be done in the following section as the visual alert icon does not have alt text AND no specific error or alert is announced.

## Acceptance Criteria

- [ ] Recommended, upon switching their selection to "Yes, I want to upload my document now" and clicking "update page" the user should receive a visual alert that announces the need of completing the following section.
- [x] Band-aid, provide alternative text to the visual warning icons using `aria-describedby`.
-->

## Steps to Recreate

1. Select any choice but "yes, I want to upload my document now" and reach the review page
2. On the review page using a screen reader, change your choice to "yes, I want to upload my document now"
3. Confirm that visually alert icons appear, but the screen reader announces nothing
4. Confirm that you are unable to access the alert icons (since they have the `role="presentation"`)

### Recommended, provide a visible alert informing users that they must complete the following section.
Provide an error message or alert informing the user that they must complete the "upload your legal documentation section" upon pressing the "update page" button for representative documentation.
<img width="1540" alt="Screen Shot 2021-05-10 at 11 40 42 PM" src="https://user-images.githubusercontent.com/14154792/117754892-3c2cd700-b1e9-11eb-99ff-b3f2352dbd80.png">

### Bandaid, provide alt text for the warning icon and use `aria-describedby` to link it to the button heading
<img width="1591" alt="Screen Shot 2021-05-10 at 11 31 27 PM" src="https://user-images.githubusercontent.com/14154792/117754318-18b55c80-b1e8-11eb-8422-ec634ab6a79e.png">

## WCAG or Vendor Guidance (optional)
* [WCAG 3.3.1 – Error Identification (Level A)](https://www.wuhcag.com/error-identification/)

## Screenshots or Trace Logs
<img width="1530" alt="Screen Shot 2021-05-10 at 11 05 45 PM" src="https://user-images.githubusercontent.com/14154792/117754975-67172b00-b1e9-11eb-9d73-612b5677cf97.png">
<img width="800" alt="Screen Shot 2021-05-10 at 10 56 28 PM" src="https://user-images.githubusercontent.com/14154792/117754992-6bdbdf00-b1e9-11eb-92ab-7895801d5330.png">

https://user-images.githubusercontent.com/14154792/117756020-5ff11c80-b1eb-11eb-8863-4c5da3f42d9a.mov

**Comment by obliviga** [comment]
Hi @joshkimux, I'm only seeing two sections in the review section of the 10-10ez form:
![image.png](https://images.zenhubusercontent.com/6283eaffdea4c21364d67e5a/4c86d4cf-a8d0-484e-a498-17e9da90c9f8)

Are you able to provide steps to reproduce how it shows up in the screenshots provided in the ticket description? Thank you!

**Comment by obliviga** [comment]
FYI, the band-aid fix mentioned in this ticket has been applied in PR #21472.

The band-aid fix is being tracked in the following ticket: 
https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/

cc @taylorkaren

**Comment by obliviga** [comment]
Hi @cohnjesse, would you happen to know how I could reproduce the UI mentioned in my previous comment? https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/276#issuecomment-1165882112

Thank you!

**Comment by joshkimux** [comment]
@obliviga , I believe I've found it: it's 10-10cg; still struggling to figure out how to replicate this as I can't seem to get to the file upload page. [The staging review documentation is here.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26697)

<img width="935" alt="Screen Shot 2022-06-28 at 8 58 34 AM" src="https://user-images.githubusercontent.com/14154792/176184361-d34251f2-c1a0-4c2c-85c0-c86e88c84e4e.png">

**Comment by taylorkaren** [comment]
Anish reaching out to Jesse Cohn about this.

**Comment by lzapson** [comment]
@joshkimux @taylorkaren The 10-10CG form has changed for application signature and upload a document. In Staging you can get to the upload a document page by following the steps below. 

- Go to https://staging.va.gov/family-member-benefits/apply-for-caregiver-assistance-form-10-10cg/introduction
- click on start your application
- enter in veteran information (WESLEY FORD, SSN: 796043735, birth date:  Tue, 06 May 1986)
- enter veteran address
- enter  recent medical care
- answer yes to Would you like to apply for benefits for a Primary Family Caregiver
- enter primary caregiver information
- enter primary caregiver address
- answer no to Health care coverage
- answer no to Would you like to apply for benefits for a Secondary Family Caregiver
- select A representative on the application signature page
-  upload document
![image.png](https://images.zenhubusercontent.com/6195603c38ef1a28570c4f7b/325ad781-7aee-4f37-a07a-899cae8142c2)
![image.png](https://images.zenhubusercontent.com/6195603c38ef1a28570c4f7b/7592a381-bdc2-4f61-ab0e-f95f861e449d)

**Comment by obliviga** [comment]
Verified in Staging: ![image.png](https://images.zenhubusercontent.com/6283eaffdea4c21364d67e5a/bbbcf14d-7c1c-4afa-9f51-7ce257d1d02c)

**Comment by obliviga** [comment]
@joshkimux After testing is complete, feel free to create another ticket if applicable. Thank you!

**Comment by joshkimux** [comment]
@obliviga , [I just tested this and wrote up a follow up ticket documenting an issue](https://github.com/department-of-veterans-affairs/va-forms-system-core/issues/474)! Let me know if I can add any more details there to explain 🙇
